kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/MultimediaOptionsGridView.java

```diff
@@ -61,29 +61,29 @@ public void executeMethod(String key) {
             ((ConversationActivity) context).processLocation();
         } else if (key.equals(context.getString(R.string.al_camera))) {
             if (storagePermissionListener.isPermissionGranted()) {
-                ((ConversationActivity) context).isTakePhoto(true);
+                ((ConversationActivity) context).setFromCamera(true);
                 ((ConversationActivity) context).processCameraAction();
             } else {
                 storagePermissionListener.checkPermission(new KmStoragePermission() {
                     @Override
                     public void onAction(boolean didGrant) {
                         if (didGrant) {
-                            ((ConversationActivity) context).isTakePhoto(true);
+                            ((ConversationActivity) context).setFromCamera(true);
```

Comment:
yes, it will break the code. there is a if statement that uses the fromCamera in the onRequestPermissionResult callback in conversation activity. if fromCamera is false then the video recorder opens, for true the camera opens

Reply:
i have explained the answers to this and the next two comments in a separate comment at the end.

Reply:
in that case, do you think it will be better to have different return code itself, rather than storing some property somewhere and then adding check for
startActivityForResult(contentChooserIntent, CODE_HERE)

Reply:
yes, using request code now, removed all fromX booleans